AWSCRTAndroidTestRunner/app/src/main/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,80 @@
+# For more information about using CMake with Android Studio, read the
+# documentation: https://d.android.com/studio/projects/add-native-code.html
+
+# Sets the minimum version of CMake required to build the native library.
+
+cmake_minimum_required(VERSION 3.4.1)
+
+# AWS lib
+set(path_to_common "${CMAKE_CURRENT_LIST_DIR}/../../../../..")
+get_filename_component(path_to_common ${path_to_common} ABSOLUTE)
+
+# This is required in order to append /lib/cmake to each element in CMAKE_PREFIX_PATH
+set(AWS_MODULE_DIR "/${CMAKE_INSTALL_LIBDIR}/cmake")
+string(REPLACE ";" "${AWS_MODULE_DIR};" AWS_MODULE_PATH "${CMAKE_PREFIX_PATH}${AWS_MODULE_DIR}")
+# Append that generated list to the module search path
+list(APPEND CMAKE_MODULE_PATH ${AWS_MODULE_PATH})
+
+list(APPEND CMAKE_MODULE_PATH "${path_to_common}/cmake")
+include(AwsFindPackage)
+set(IN_SOURCE_BUILD ON)
+set(BUILD_SHARED_LIBS ON)
+
+# We will generate our own tests, the tests that are there depend on CTest
+set(BUILD_TESTING OFF)
+add_subdirectory(${path_to_common} ${CMAKE_CURRENT_BINARY_DIR}/aws-c-common)
+aws_use_package(aws-c-common)
+
+function(import_tests test_cmakelists)
+    # Find all test cases in tests/CMakeLists.txt
+
+    file(STRINGS ${test_cmakelists} test_case_lines REGEX "add_test_case.+")
```

Comment:
yeah this one could probably more easily be pulled by adding the calls to add_test_case to a cmake variable or something and then using that

Reply:
maybe not though.... your call.... we also have to deal with the tests that are gated by platform (like windows and apple for example)

Reply:
Did it.